### ISSUE TITLE
🐞 Can't connect to docker daemon when using nerdctl

### ISSUE THREAD

**Issue by Keyruu** [dagger/dagger#1959]
### What is the issue?

I use nerdctl with buildkit and containerd (Rancher Desktop) to normally build, tag and push my "docker" images (OCI compatible). If I want to use the docker Dagger step as mentioned in the "quick start" [here](https://docs.dagger.io/), I get the error that it couldn't connect to the docker daemon. I was told to create this issue by helderco#4068 in Discord.

### Log output

`11:09AM ERR failed to run docker: exit status 1    output=Client:
 Context:    default
11:09AM ERR failed to run docker: exit status 1    output=Client:
 Context:    default
 Debug Mode: falseuildx (Docker Inc., v0.7.1)
 Plugins:: Docker Compose (Docker Inc., v2.2.3)
  buildx: Docker Buildx (Docker Inc., v0.7.1)
  compose: Docker Compose (Docker Inc., v2.2.3)
  scan: Docker Scan (Docker Inc., v0.16.0)
ERROR: Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?
Server:pretty printing info
ERROR: Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?
errors pretty printing info`

### Steps to reproduce

1. Use Rancher Desktop and get use nerdctl as your building tool.
2. Alias docker to nerdctl.
3. Try to build with dagger `dagger do build` in https://github.com/dagger/dagger/pkg/universe.dagger.io/examples/todoapp

### Dagger version

0.2.4

### OS version

macOS 11.6.4

**Comment by vdemeester** [comment]
Hey @Keyruu, If you use `nerdctl` with `containerd` this effectively means you do not have a docker daemon running, and thus, no socket as well. It should be technically possible to use dagger containerd instead or directly pointing to a buildkit daemon socket.

@shykes @aluzzardi I did do this approach with my [`buildkit-tekton`](https://github.com/vdemeester/buildkit-tekton/blob/main/pkg/buildkit/buildkit.go#L49) project taking inspiration from `pkg/buildkitd`, but I think dagger could do the same thing more / less. One downside might be that dagger doesn't "control" the buildkitd version in all cases.

**Comment by helderco** [comment]
There's a doc page that hasn't been ported yet but it's still relevant: https://docs.dagger.io/1013/operator-manual/

We need to document how to make dagger work in other environments, including in the error message when the local docker socket isn't available.

**Comment by jcdickinson** [comment]
I have attempted this with podman by:

```sh
podman run -d --name buildkitd --privileged moby/buildkit:latest
export BUILDKIT_HOST=podman-container://buildkitd
dagger do build
```

I get this error:

```
FTL failed to execute plan: error detecting platform failed to list workers: Unavailable: connection error: desc = "transport: Error while dialing dial podman-container: unknown network podman-container"
```

The workaround is to:

```sh
podman run -d --name buildkitd --privileged -p 1234:1234 moby/buildkit:latest --addr tcp://0
.0.0.0:1234
export BUILDKIT_HOST=tcp://localhost:1234
dagger do build
```

This seems like a hack, though.

**Comment by aluzzardi** [comment]
`podman-container://` support was merged in #2211 and will go out with tomorrow's release.
